app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -495,41 +518,29 @@ private void setUpGravatar(final Message message, final MessageHolder messageHol
             float px = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP,
                     35, resources.getDisplayMetrics());
 
-
             String url = message.getSender().getAvatarURL() + "&s=" + px;
             url = UrlHelper.addHost(url);
 
-            Picasso.with(context)
+            Glide
+                    .with(context)
                     .load(url)
                     .placeholder(android.R.drawable.stat_notify_error)
-                    .error(android.R.drawable.presence_online)
-                    .into(messageHolder.gravatar, new Callback() {
+                    .error(defaultAvatar)
+                    .listener(new RequestListener<String, GlideDrawable>() {
                         @Override
-                        public void onSuccess() {
+                        public boolean onException(Exception e, String model, Target<GlideDrawable> target, boolean isFirstResource) {
```

Comment:
How about displaying the default gravatar in this onException ?
And using a placeholder like before? 

As this method will be called on every onBindViewHolder so we need to make sure these are taking minimal time!